### DISCUSSION TITLE
NuGet v3 API in community server

### DISCUSSION THREAD

**Discussion by dtretyakov** [chocolatey/home#261] [General]
Since `choco` CLI v2 supports NuGet v3 protocol it will be great to support it in community server.

According to the documentation it looks like it works as v2 server: https://docs.chocolatey.org/en-us/community-repository/community-packages-disclaimer

> packages from the community repository (https://community.chocolatey.org/api/v2)

Any plans to upgrade community server or it was done update documentation?

**Comment by gep13** [comment]
Yes, this is something that we are going to be undertaking, but there is currently no time frame for when this will be provided.  The decision was made to first support NuGet v3 feeds within Chocolatey CLI (which means that you can then use Chocolatey CLI with repositories that do support it), and then we will provide a V3 feed on the Chocolatey Community Repository.

**Comment by dhs-rec** [comment]
@gep13, Chocolatey CLI v2.0.0 with support for API v3 has been released May 31, 2023 (five days before your reply). 15 months have passed since that release. Could you please give a quick update on the status of v3 in the community repository?

**Reply by pauby** [reply]
NuGet v3 support hasn't been added to the Chocolatey Community Repository as yet. While this is a priority for us, it's in planning. The v2 API continues to provide Chocolatey CLI with all the functionality it needs.

**Comment by dhs-rec** [comment]
Thanks a lot.

**Comment by dhs-rec** [comment]
There's one problem with this: The documentation mentions that people can use Nexus to cache the community repository, but according to [Nexus documentation](https://help.sonatype.com/en/nuget-proxy-repositories.html), it doesn't support everything needed by Chocolatey in its NuGet V2 implementation.

**Reply by pauby** [reply]
Can you tell me what is broken?

**Reply by dhs-rec** [reply]
Sorry, forget about this. That was just my interpretation of the Nexus documentation. Did a quick test setup and it seems to work.

**Reply by pauby** [reply]
I am putting together an issue around Chocolatey CLI and Nexus just now. I will post the link to it here once it's complete.

**Reply by pauby** [reply]
Issue [created here](https://github.com/chocolatey/choco/issues/3506).

**Comment by misse** [comment]
Hello folks, any update on this issue? We're being rate limited despite our best efforts since the V2 APIs metadata is impossible to cache according to Jfrog

**Reply by Windos** [reply]
I cannot speak to any changes to the Chocolatey Community Repository (CCR), but can say that yes, it appears that JFrog Artifactory cannot/doesn't cache any metadata when used as a proxy/remote repo between clients and the CCR. I have a public ticket open here about this: https://jfrog.atlassian.net/browse/RTFACT-31023

It should be noted that both Sonatype Nexus Repository and Inedo ProGet are able to function as a caching repository and help to prevent hitting rate limits or being blocked for excessive usage.

**Comment by ctJanF** [comment]
Hello. 
Are there any new updates on this topic? It's been almost 2 1/2 years now. In addition to our internal repo (v3), we also use a Choco proxy. Because of this, we cannot activate the “usePackageRepositoryOptimizations” option, as otherwise users would not be able to access the public repo (via proxy). As a result, our log files are filling up with a lot of data that we would not have if we could activate the feature.

**Reply by gep13** [reply]
@ctJanF thank you for reaching out about this topic.

This is still a high priority task for us here at Chocolatey, but I am not in a position to provide any further information currently.

As you can perhaps imagine, there are quite a few moving parts when it comes to providing a v3 feed on the Chocolatey Community Repository, whilst ensuring things continue to work for things like the Package Moderation Services.

There have been several internal discussions around this topic, but no information about _when_ this will be available.
